src/game/logic/object/weaponset.h

```diff
@@ -150,8 +176,13 @@ class WeaponSet : public SnapShot
     const Weapon *Get_Cur_Weapon() const { return m_weapons[m_curWeapon]; }
     Weapon *Get_Cur_Weapon() { return m_weapons[m_curWeapon]; }
 
+#ifdef GAME_DLL
+    WeaponSet *Hook_Ctor() { return new (this) WeaponSet(); }
+    void Hook_Dtor() { WeaponSet::~WeaponSet(); }
+#endif
+
 private:
-    WeaponTemplateSet *m_curWeaponTemplateSet;
+    const WeaponTemplateSet *m_curWeaponTemplateSet;
     Weapon *m_weapons[WEAPONSLOT_COUNT];
     WeaponSlotType m_curWeapon;
     int m_curWeaponLocked;
```

Comment:
Should m_curWeaponLocked be also of type WeaponSlotType? The xfer function call indicates so.

Reply:
fixed.